tests/bsim/bluetooth/audio/src/common.c

```diff
@@ -135,3 +140,181 @@ void test_init(void)
 	bst_ticker_set_next_tick_absolute(WAIT_TIME);
 	bst_result = In_progress;
 }
+
+#define SYNC_MSG_SIZE 1
+static int32_t dev_cnt;
+static uint backchannel_nums[255];
+static uint chan_cnt;
+static void register_more_cmd_args(void)
+{
+	static bs_args_struct_t args_struct_toadd[] = {
+		{
+			.option = "D",
+			.name = "number_devices",
+			.type = 'i',
+			.dest = (void *)&dev_cnt,
+			.descript = "Number of devices which will connect in this phy",
+			.is_mandatory = true,
+		},
+		ARG_TABLE_ENDMARKER,
+	};
+
+	bs_add_extra_dynargs(args_struct_toadd);
+}
+NATIVE_TASK(register_more_cmd_args, PRE_BOOT_1, 100);
+
+/**
+ * @brief Get the channel id based on remote device ID
+ *
+ * The is effectively a very simple hashing function to generate unique channel IDs from device IDs
+ *
+ * @param dev 16-bit device ID
+ *
+ * @return uint 32-bit channel ID
+ */
+static uint get_chan_num(uint16_t dev)
+{
+	uint16_t self = (uint16_t)bsim_args_get_global_device_nbr();
+	uint channel_id;
+
+	if (self < dev) {
+		channel_id = (dev << 16) | self;
+	} else {
+		channel_id = (self << 16) | dev;
+	}
+
+	return channel_id;
+}
+
+/**
+ * @brief Set the up backchannels between each pair of device
+ *
+ * Each pair of devices will get a unique channel
+ */
+static void setup_backchannels(void)
+{
+	__ASSERT_NO_MSG(dev_cnt > 0 && dev_cnt < ARRAY_SIZE(backchannel_nums));
+	const uint self = bsim_args_get_global_device_nbr();
+	uint device_numbers[dev_cnt];
+	uint *channels;
+
+	for (int32_t i = 0; i < dev_cnt; i++) {
+		if (i != self) { /* skip ourselves*/
+			backchannel_nums[chan_cnt] = get_chan_num((uint16_t)i);
+			device_numbers[chan_cnt++] = i;
+		}
+	}
+
+	channels = bs_open_back_channel(self, device_numbers, backchannel_nums, chan_cnt);
+	__ASSERT_NO_MSG(channels != NULL);
+}
+NATIVE_TASK(setup_backchannels, PRE_BOOT_3, 100);
```

Comment:
Wouldn't it make sense to factor out this piece of code from `tests/bsim/bluetooth/host/att/long_read/testlib/bs_sync.c` instead of duplicating the code? Applies to `register_more_cmd_args` as well

Reply:
This code is similar, but not identical. 

The code in `tests/bsim/bluetooth/host/att/long_read/testlib/bs_sync.c` sets up a single channel for all devices, where this sets up a channel per pair of devices. 

So for 3 devices, this sets up
Channel between dev 0 and dev 1
Channel between dev 0 and dev 2
Channel between dev 1 and dev 2

Reply:
Besides that, we have no current way of sharing code between BSIM tests - Currently they are considered separate, and something like this would be probably be placed in a common framework at some point.